service/arm_templates/network_setup.json

```diff
@@ -0,0 +1,100 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "virtualNetworksAPIVersion": {
+      "type": "string",
+      "defaultValue": "2017-03-01",
+      "metadata": {
+        "description": "API version used by the Microsoft.Network/virtualNetworks resource."
+      }
+    },
+    "addressPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.0.0/16",
+      "metadata": {
+        "description": "The main address block reserved for this virtual network."
+      }
+    },
+    "subnetMasterCIDR": {
+      "type": "string",
+      "defaultValue": "10.0.1.0/24",
+      "metadata": {
+        "description": "The CIDR block for the Kubernetes masters' subnet"
+      }
+    },
+    "subnetWorkerCIDR": {
+      "type": "string",
+      "defaultValue": "10.0.2.0/24",
+      "metadata": {
+        "description": "The CIDR block for the Kubernetes workers' subnet"
+      }
+    },
+    "mastersSecurityGroupID": {
+      "type": "string"
+    },
+    "workersSecurityGroupID": {
+      "type": "string"
+    }
+  },
+  "variables": {
+    "virtualNetworkName": "[concat(uniquestring(resourceGroup().id), '-VirtualNetwork')]",
```

Comment:
I think it would be better if we used the cluster ID here rather than `uniquestring`. The cluster IDs are unique within an installation and it looks like Virtual Network names don't have to be globally unique.

https://docs.microsoft.com/en-us/azure/architecture/best-practices/naming-conventions

I'd like our resource names to be predictable and driven by the cluster TPO as much as possible. This should mean we need fewer output variables from the templates.